src/core/init/media_source_content_initializer.ts

```diff
@@ -523,9 +523,15 @@ export default class MediaSourceContentInitializer extends ContentInitializer {
      */
     function handleStreamOrchestratorCallbacks() : IStreamOrchestratorCallbacks {
       return {
-        needsBufferFlush: () => {
-          const seekedTime = mediaElement.currentTime + 0.001;
-          playbackObserver.setCurrentTime(seekedTime);
+        needsBufferFlush: (relativeResumingPosition?) => {
+          let wantedSeekingTime: number;
+          const currentTime = playbackObserver.getCurrentTime();
+          if (relativeResumingPosition !== undefined && relativeResumingPosition !== 0) {
+            wantedSeekingTime = currentTime + relativeResumingPosition;
+          } else {
+            wantedSeekingTime = currentTime + 0.001;
```

Comment:
Here we would seek 1 millisecond forward if set to `0`.

Maybe we should make a difference between the RxPlayer's default and an API `0` value, which would indicate to seek in-place?